src/Components/Votes/Votes.jsx

```diff
@@ -0,0 +1,50 @@
+import { incrementVotes, decrementVotes } from "../../Utils/api";
+import { useState } from "react";
+import "./Votes.css";
+
+function Votes({ article, setArticle, article_id }) {
+  const [voted, setVoted] = useState(false);
+
+  function handleClick() {
+    if (voted === false) {
+      setVoted(true);
+      setArticle((article) => {
+        return { ...article, votes: article.votes + 1 };
+      });
+      incrementVotes(article_id)
+        .then(() => {
+          return { ...article, votes: article.votes - 1 };
+        })
+        .catch((err) => {
+          if (err) {
```

Comment:
we'd also at this point want to undo our optimistic rendering as the vote failed to be added to the db but we've already shown our user that the vote has been added